### DISCUSSION TITLE
Proper way to ensure that a Q-Select has an option selected if options exist?

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#15398] [General - Components / Directives / etc] [answered]
Silly question, but I'm unsure how to approach this simple issue.  

If options exist for a Q-Select, I always want an option selected (the first one by default). I never want the Q-Select to have no value.. I will just disable it if no values are available to pick from.

I tried to set up a watcher for this and it doesn't update the Q-Select.  It sets the v-model as it should, but it doesn't visually affect the component in any way.  (options is an array of objects)

```js
<q-select v-model="selectedOption" :options="options" option-label="name" emit-value ... />

watch: {
  options: function () {
    if (this.selectedOption === null) {
      if (this.options.length > 0) {
        this.selectedOption = this.options[0];
      }
    }
  },
},
```

What is the "correct" way to approach this?

**Comment by yusufkandemir** [comment, marked as answer]
It's almost crucial to create a Codepen for questions like this. So, please try to share one next time.

From what I can see, you are using `emit-value`, so the model will not be the object itself, it will be the value of one of its properties(_`value` by default, code example is not complete, so unsure_). So, you should do `this.selectedOption = this.options[0].value`(_or the property you set in `option-value`_). If you are not using `map-options`, you will only see the value, not the label. So, you might want to use that as well.

**Reply by ghost** [reply]
Thank you for the help.  I had a number of issues going on all at once, it turns out.  

* I was missing map-options; thanks for pointing that out.
* I was using the whole object as the model, so I fixed that.
* For some reason, `option-label="name"` still doesn't work, and I'm forced to use `:option-label="(opt) => (opt === null ? '' : opt.name)"` for a value.  `this.selectedOption = this.options[0].value` selects the item but the label never updates unless I configure the option-label in this manner.  I don't have a clue why, and it took me forever to figure this out.
* I was using the selected-item slot for custom display features, which was causing additional issues on top of everything else, so for now I have just disabled that in favor of :option-label="..." as described above.

I still have more to do to get this working smoothly but at least it's working now. I've spent more time wrestling with the QSelect component than anything else in my project.  I'll mark as resolved and continue working through it.  Thanks again.
